delivery/app.py

```diff
@@ -21,38 +59,37 @@ def __init__(self, restaurants):
         self.restaurants = restaurants
 
     def get(self):
-        return jsonify(self.restaurants), 200
+        return jsonify(self.restaurants.get_all()), 200
 
     def post(self):
         content = request.json
         schema = RestaurantCreateOrUpdateSchema()
         schema.load(content)
-        content['id'] = self.restaurants[len(self.restaurants) - 1]['id'] + 1
-        self.restaurants.append(content)
-        return jsonify(self.restaurants), 201
+        id = len(self.restaurants.get_all())
+        content['id'] = id
+        self.restaurants.save(content)
+        return jsonify(self.restaurants.get_all()), 201
 
 
 class RestaurantItemEndpoint(MethodView):
     def __init__(self, restaurants):
         self.restaurants = restaurants
 
     def get(self, restaurant_id):
-        for restaurant in self.restaurants:
-
-            if restaurant["id"] == restaurant_id:
-                return jsonify(restaurant), 200
-        raise WrongIdError(description="Restaurant with such ID doesn't exist")
+        try:
+            return jsonify(self.restaurants.get_by_id(restaurant_id)), 200
+        except IndexError:
+            raise WrongIdError(description="Restaurant with such ID doesn't exist")
 
     def put(self, restaurant_id):
         content = request.json
         schema = RestaurantCreateOrUpdateSchema(partial=True)
         schema.load(content)
-        for restaurant in self.restaurants:
-            if restaurant["id"] == restaurant_id:
-                for key, value in content.items():
-                    restaurant[key] = value
-                return jsonify(restaurant), 200
-        raise WrongIdError(description="No restaurant to update. Restaurant with such ID doesn't exist")
+        self.restaurants.save(content, restaurant_id)
+        try:
+            return jsonify(self.restaurants.get_by_id(restaurant_id)), 200
+        except IndexError:
+            raise WrongIdError(description="No restaurant to update. Restaurant with such ID doesn't exist")
```

Comment:
different repository implementations will have different exceptions, here is it `IndexError` but Mongo might have its `MongoError`. it is better to do validation inside repository code, i.e. `save()` will raise `WrongIdError`

Reply:
fixed